Upstream-Status: Inappropriate No time for upstreaming 

--- a/drivers/rtc/rtc-rv3028.c
+++ b/drivers/rtc/rtc-rv3028.c
@@ -245,9 +245,16 @@
 static int rv3028_update_cfg(struct rv3028_data *rv3028, unsigned int reg,
 			     unsigned int mask, unsigned int val)
 {
-	u32 eerd;
+	u32 eerd, cur;
 	int ret;
 
+	ret = regmap_read(rv3028->regmap, reg, &cur);
+	if (ret)
+		return ret;
+
+	if ((cur & mask) == (val & mask))
+		return 0;
+
 	ret = rv3028_enter_eerd(rv3028, &eerd);
 	if (ret)
 		return ret;
