Upstream-Status: Inappropriate [Yocto native integration adjustments]
---
--- a/spsdk/uboot/spsdk_uuu.py	2026-01-30 08:17:19.000000000 -0500
+++ b/spsdk/uboot/spsdk_uuu.py	2026-03-26 15:55:31.183800089 -0400
@@ -19,8 +19,11 @@
 from typing import Any, Callable, Optional, no_type_check
 
 import click
-from libuuu import LibUUU, UUUNotifyCallback, UUUState
-from libuuu.libuuu import UUUNotifyStruct, UUUNotifyType, _default_notify_callback
+try:
+    from libuuu import LibUUU, UUUNotifyCallback, UUUState
+    from libuuu.libuuu import UUUNotifyStruct, UUUNotifyType, _default_notify_callback
+except ImportError:
+    pass
 
 from spsdk.exceptions import SPSDKError, SPSDKValueError
 from spsdk.utils.database import DatabaseManager, UsbId
@@ -43,8 +46,7 @@
 )
 
 
-@UUUNotifyCallback
-def _spsdk_notify_callback(struct: UUUNotifyStruct, data) -> int:  # type: ignore
+def _spsdk_notify_callback(struct, data) -> int:  # type: ignore
     """Default callback function that stores UUU response in class variable.
 
     This callback processes UUU notifications and updates the internal state. For command info
@@ -61,7 +63,7 @@
     return 1 if SPSDKUUU._state.error else 0
 
 
-class SPSDKUUUState(UUUState):
+class SPSDKUUUState:
     """SPSDK UUU State Manager.
 
     This class extends UUUState to provide enhanced state management for UUU operations
@@ -112,7 +114,7 @@
             # Enable cursor again
             click.echo("\033[?25h", nl=False)
 
-    def update(self, struct: UUUNotifyStruct) -> None:
+    def update(self, struct) -> None:
         """Update the state with a notification from uuu.
 
         This method processes different types of notifications from the uuu tool and updates
--- a/spsdk/utils/devicedescription.py	2026-01-30 08:17:19.000000000 -0500
+++ b/spsdk/utils/devicedescription.py	2026-03-26 15:47:00.502849229 -0400
@@ -15,7 +15,10 @@
 from abc import ABC, abstractmethod
 from typing import Optional, Union
 
-from libusbsio.libusbsio import LIBUSBSIO
+try:
+    from libusbsio.libusbsio import LIBUSBSIO
+except ImportError:
+    pass
 
 from spsdk.mboot.interfaces.usb import MbootUSBInterface
 from spsdk.sdp.interfaces.usb import SdpUSBInterface
--- a/spsdk/utils/interfaces/device/usb_device.py	2026-01-30 08:17:19.000000000 -0500
+++ b/spsdk/utils/interfaces/device/usb_device.py	2026-03-26 15:47:00.503849240 -0400
@@ -14,7 +14,10 @@
 import logging
 from typing import Optional
 
-import libusbsio
+try:
+    import libusbsio
+except ImportError:
+    pass
 from typing_extensions import Self
 
 from spsdk.exceptions import SPSDKConnectionError, SPSDKError
--- a/spsdk/utils/interfaces/device/usbsio_device.py	2026-01-30 08:17:19.000000000 -0500
+++ b/spsdk/utils/interfaces/device/usbsio_device.py	2026-03-26 15:47:00.503849240 -0400
@@ -17,8 +17,11 @@
 from dataclasses import dataclass
 from typing import Optional, Union
 
-import libusbsio
-from libusbsio.libusbsio import LIBUSBSIO
+try:
+    import libusbsio
+    from libusbsio.libusbsio import LIBUSBSIO
+except ImportError:
+    pass
 from typing_extensions import Self
 
 from spsdk.exceptions import SPSDKConnectionError, SPSDKError, SPSDKValueError
@@ -396,7 +399,7 @@
         return (args, kwargs)
 
     @staticmethod
-    def _get_usbsio() -> LIBUSBSIO:
+    def _get_usbsio():
         """Get USBSIO library instance with SPSDK error handling.
 
         Wraps the libusbsio library initialization to provide consistent SPSDK error handling
@@ -426,7 +429,7 @@
         :return: List of port indexes of found USBSIO devices
         """
 
-        def _filter_usb(sio: LIBUSBSIO, ports: list[int], flt: str) -> list[int]:
+        def _filter_usb(sio, ports: list[int], flt: str) -> list[int]:
             """Filter LIBUSBSIO devices based on provided criteria.
 
             Filters the list of available LIBUSBSIO ports using the specified filter string which can
--- a/spsdk/utils/nxpdevscan.py	2026-01-30 08:17:19.000000000 -0500
+++ b/spsdk/utils/nxpdevscan.py	2026-03-26 15:47:00.503849240 -0400
@@ -19,7 +19,10 @@
 import struct
 from typing import Any, Optional
 
-from libusbsio import LIBUSBSIO_Exception, usbsio
+try:
+    from libusbsio import LIBUSBSIO_Exception, usbsio
+except ImportError:
+    pass
 from serial import SerialException
 from serial.tools.list_ports import comports
 from serial.tools.list_ports_common import ListPortInfo
