--- a/btlru/src/lrdmbt.c	2021-03-03 17:18:11.000000000 
+++ b/btlru/src/lrdmbt.c	2021-03-03 18:00:17.000000000 
@@ -597,7 +597,7 @@
 	hex_dump(DBG_DEBUG, (char*)&eventBuf, rspLen);
 
 	if (eventBuf[1] == HCI_EVENT_VENDOR_SPECIFIC && eventBuf[3] != 0xC0) {
-		read(ctx->hci_sock_bz, dummy_buf, EVENT_BUF_SIZE);
+		if (read(ctx->hci_sock_bz, dummy_buf, EVENT_BUF_SIZE));
 	}
 
 	if (rsp && rsp->len) {
