--- a/drivers/mmc/core/Kconfig	2021-03-03 11:31:07.000000000 
+++ b/drivers/mmc/core/Kconfig	2021-03-03 11:41:53.000000000 
@@ -15,7 +15,7 @@
 
 config PWRSEQ_SD8787
 	tristate "HW reset support for SD8787 BT + Wifi module"
-	depends on OF && (MWIFIEX || BT_MRVL_SDIO || LIBERTAS_SDIO)
+	depends on OF
 	help
 	  This selects hardware reset support for the SD8787 BT + Wifi
 	  module. By default this option is set to n.
