
Upstream-Status: Inappropriate No time for upstreaming

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index b189aab8dd94..2fedbe3d0ee5 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -1300,6 +1300,9 @@ static int sdma_load_context(struct sdma_channel *sdmac)
 	int ret;
 	unsigned long flags;
 
+	if (!bd0)
+		return -EBUSY;
+
 	if (sdmac->direction == DMA_DEV_TO_MEM)
 		load_address = sdmac->pc_from_device;
 	else if (sdmac->direction == DMA_DEV_TO_DEV)
