
Upstream-Status: Inappropriate No time for upstreaming 

diff --git a/drivers/rtc/rtc-rv3028.c b/drivers/rtc/rtc-rv3028.c
index c2a531f0e125..d1887048fc85 100644
--- a/drivers/rtc/rtc-rv3028.c
+++ b/drivers/rtc/rtc-rv3028.c
@@ -1021,6 +1021,8 @@ static int rv3028_probe(struct i2c_client *client)
 	nvmem_cfg.priv = rv3028->regmap;
 	devm_rtc_nvmem_register(rv3028->rtc, &nvmem_cfg);
 	eeprom_cfg.priv = rv3028;
+	eeprom_cfg.of_node = of_get_compatible_child(client->dev.of_node,
+		"microcrystal,rv3028-eeprom");
 	devm_rtc_nvmem_register(rv3028->rtc, &eeprom_cfg);
 
 	rv3028->rtc->max_user_freq = 1;
