Added support for whatever mode firmware version 0098_v00_T4 is at

Upstream-Status: Inappropriate No time for upstreaming

diff --git a/drivers/input/touchscreen/exc3000.c b/drivers/input/touchscreen/exc3000.c
index 8f6205dfc95d..99253ad5abf9 100644
--- a/drivers/input/touchscreen/exc3000.c
+++ b/drivers/input/touchscreen/exc3000.c
@@ -141,10 +141,12 @@ static int exc3000_read_frame(struct exc3000_data *data, u8 *buf)
 	if (ret != EXC3000_LEN_FRAME)
 		return -EIO;
 
-	if (get_unaligned_le16(buf) != EXC3000_LEN_FRAME)
+	ret = get_unaligned_le16(buf);
+
+	if (ret > EXC3000_LEN_FRAME)
 		return -EINVAL;
 
-	return 0;
+	return ret;
 }
 
 static int exc3000_handle_mt_event(struct exc3000_data *data)
@@ -201,7 +203,7 @@ static irqreturn_t exc3000_interrupt(int irq, void *dev_id)
 	int ret;
 
 	ret = exc3000_read_frame(data, buf);
-	if (ret) {
+	if (ret < 0) {
 		/* Schedule a timer to release "stuck" contacts */
 		exc3000_schedule_timer(data);
 		goto out;
