
Upstream-Status: Inappropriate NXP proprietary driver

diff --git a/drivers/gpu/drm/imx/dw_mipi_dsi-imx.c b/drivers/gpu/drm/imx/dw_mipi_dsi-imx.c
index be9c584ccafd..139ababf4fa5 100644
--- a/drivers/gpu/drm/imx/dw_mipi_dsi-imx.c
+++ b/drivers/gpu/drm/imx/dw_mipi_dsi-imx.c
@@ -77,30 +77,6 @@ static bool dw_mipi_dsi_phy_validate_helper(void *priv_data,
 
 	return true;
 }
-
-static bool dw_mipi_dsi_imx_is_mode_clock_valid(int clock)
-{
-	int i;
-	const int valid_mode_clocks[] = {
-		25175, 25200, 27000, 27027, 28320,
-		31500,
-		40000, 49500,
-		50000, 54000, 54054, 59341, 59400,
-		65000,
-		74176, 74250, 78750,
-		108000, 108108,
-		132000,
-		148352, 148500,
-	};
-
-	for (i = 0; i < ARRAY_SIZE(valid_mode_clocks); i++) {
-		if (clock == valid_mode_clocks[i])
-			return true;
-	}
-
-	return false;
-}
-
 static enum drm_mode_status
 __dw_mipi_dsi_imx_mode_valid(void *priv_data,
 			     const struct drm_display_mode *mode,
@@ -114,9 +90,6 @@ __dw_mipi_dsi_imx_mode_valid(void *priv_data,
 	unsigned long pixel_clock_rate = mode->clock * 1000;
 	unsigned long adjusted_clock_rate;
 
-	if (!dw_mipi_dsi_imx_is_mode_clock_valid(mode->clock))
-		return MODE_ERROR;
-
 	ret = dw_mipi_dsi_phy_validate_helper(priv_data,
 					      &bpp,
 					      mode,
