diff --git a/sound/soc/fsl/Kconfig b/sound/soc/fsl/Kconfig
--- a/sound/soc/fsl/Kconfig
+++ b/sound/soc/fsl/Kconfig
@@ -544,7 +544,7 @@ config SND_SOC_IMX_HDMI
 
 config SND_SOC_IMX_CDNHDMI
 	tristate "SoC Audio support for i.MX boards with CDN HDMI port"
-	depends on DRM_IMX_CDNS_MHDP
+	depends on DRM_IMX_CDNS_MHDP || DRM_IMX_HDMI
 	select SND_SOC_IMX_PCM_DMA
 	select SND_SOC_FSL_SAI
 	select SND_SOC_HDMI_CODEC

