diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index 5e3253cfc790..a1a288c40b05 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -1343,8 +1343,8 @@ static int fsl_sai_probe(struct platform_device *pdev)
 	/* No error out for old DTB cases but only mark the clock NULL */
 	sai->bus_clk = devm_clk_get(&pdev->dev, "bus");
 	if (IS_ERR(sai->bus_clk)) {
-		dev_err(&pdev->dev, "failed to get bus clock: %ld\n",
-				PTR_ERR(sai->bus_clk));
+		dev_err_probe(&pdev->dev, PTR_ERR(sai->bus_clk),
+			"failed to get bus clock\n");
 		return PTR_ERR(sai->bus_clk);
 		sai->bus_clk = NULL;
 	}
