
Upstream-Status: Inappropriate No time for upstreaming

--- a/arch/arm/mach-imx/imx8m/Kconfig
+++ b/arch/arm/mach-imx/imx8m/Kconfig
@@ -445,6 +445,17 @@ config TARGET_LIBREM5
 	select IMX8MQ
	select SUPPORT_SPL
	select IMX8M_LPDDR4
+
+config TARGET_IMX8MP_SUMMITSOM
+	bool "Summit SOM i.MX8M Plus"
+	select IMX8MP
+	select SUPPORT_SPL
+	select IMX8M_LPDDR4
+	select FSL_CAAM
+	select FSL_BLOB
+	select MISC
+	select ARCH_MISC_INIT
+	select SPL_CRYPTO if SPL
 
 endchoice
 
@@ -481,5 +492,6 @@ source "board/technexion/pico-imx8mq/Kconfig"
 source "board/variscite/imx8mn_var_som/Kconfig"
 source "board/toradex/verdin-imx8mm/Kconfig"
 source "board/toradex/verdin-imx8mp/Kconfig"
+source "board/summit/imx8mp_summitsom/Kconfig"
 
 endif
