--- a/handlers/raw_handler.c
+++ b/handlers/raw_handler.c
@@ -170,8 +170,8 @@
 			return -1;
 		}
 
-		if (snprintf(path, sizeof(path), "%s%s",
-					 DATADST_DIR, img->path) >= (int)sizeof(path)) {
+		if (snprintf(path, sizeof(path), "%s%s%s",
+			DATADST_DIR, img->path[0] != '/' ? "/" : "", img->path) >= (int)sizeof(path)) {
 			ERROR("Path too long: %s%s", DATADST_DIR, img->path);
 			goto cleanup;
 		}
